hyperspy/signal.py

```diff
@@ -1973,12 +1980,17 @@ def plot(self, navigator="auto", axes_manager=None, plot_markers=True,
 
         self._plot.axes_manager = axes_manager
         self._plot.signal_data_function = self.__call__
+
+        if self.metadata.has_item("Signal.quantity"):
+            self._plot.quantity_label = self.metadata.Signal.quantity
         if self.metadata.General.title:
-            self._plot.signal_title = self.metadata.General.title
+            title = self.metadata.General.title
+            if kwargs.get('power_spectrum', False):
+                title = title.replace('FFT', 'Power spectrum')
```

Comment:
This assumes that FFT is contained in the title. It may not be the case because the user has changed it or the data comes from an external source.